pkg/pool-utils/contracts/factories/BasePoolFactory.sol

```diff
@@ -35,7 +37,12 @@ import "@balancer-labs/v2-solidity-utils/contracts/helpers/SingletonAuthenticati
  * become increasingly important. Governance can deprecate a factory by calling `disable`, which will permanently
  * prevent the creation of any future pools from the factory.
  */
-abstract contract BasePoolFactory is IBasePoolFactory, BaseSplitCodeFactory, SingletonAuthentication {
+abstract contract BasePoolFactory is
+    IBasePoolFactory,
+    BaseSplitCodeFactory,
+    SingletonAuthentication,
+    FactoryWidePauseWindow
```

Comment:
I would've made this change (bundling factory-wide into basepoolfactory) in a separate PR, since it is entirely unrelated to configurable delays, and greatly pollutes the diff, but oh well.